### PR TITLE
Remove allocations from executable memory pool

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Remove allocations from executable memory pool

Resolves: #237 

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

File: libs/platform/ebpf_epoch.h [hunk 1]
```diff
@@ -52,7 +52,7 @@ extern "C"
      * @returns Pointer to memory block allocated, or null on failure.
      */
     void*
-    ebpf_epoch_allocate(size_t size, ebpf_memory_type_t type);
```
Comment (dthaler) [inline libs/platform/ebpf_epoch.h:55]: delete line 51

File: libs/platform/ebpf_trampoline.c [hunk 2]
```diff
@@ -6,42 +6,125 @@
...
 #endif
 }
+
+ebpf_result_t
+ebpf_get_trampoline_function(const ebpf_trampoline_table_t* trampoline_table, size_t index, void** function)
+{
+    ebpf_trampoline_entry_t* local_entries;
+    ebpf_result_t return_value;
+
+    if (index > trampoline_table->entry_count) {
```
Comment (dthaler) [inline libs/platform/ebpf_trampoline.c:113]: shouldn't this be `>=`  instead of `>` ?

File: libs/platform/kernel/ebpf_platform_kernel.c [hunk 3]
```diff
@@ -31,10 +31,9 @@ ebpf_platform_terminate()
 {}
 
 void*
-ebpf_allocate(size_t size, ebpf_memory_type_t type)
+ebpf_allocate(size_t size)
 {
-    return ExAllocatePool2(
-        type == EBPF_MEMORY_EXECUTE ? POOL_FLAG_NON_PAGED_EXECUTE : POOL_FLAG_NON_PAGED, size, EBPF_POOL_TAG);
+    return ExAllocatePool2(POOL_FLAG_NON_PAGED_EXECUTE, size, EBPF_POOL_TAG);
```
Comment (dthaler) [inline libs/platform/kernel/ebpf_platform_kernel.c:36]: Before, almost all callers passed EBPF_MEMORY_NO_EXECUTE which would result in POOL_FLAG_NON_PAGED, so why the change to return executable instead?   The title of the PR talks about removing allocations from executable pool, so I expected this line to say
```suggestion
    return ExAllocatePool2(POOL_FLAG_NON_PAGED, size, EBPF_POOL_TAG);
```

Comment (Alan-Jowett) [reply libs/platform/kernel/ebpf_platform_kernel.c]: Good catch. Yes, I removed the wrong branch.

**Review by dthaler — APPROVED** [review]
Looks good to me now
